### PR TITLE
[api] change friendly name to 'const char *'

### PR DESCRIPTION
**Pull request by lewurm**

### CODE REVIEW

File: mono/metadata/appdomain.h [hunk 1]
```diff
@@ -67,7 +67,7 @@ mono_domain_create         (void);
 
 MONO_RT_EXTERNAL_ONLY
 MONO_API MonoDomain *
-mono_domain_create_appdomain (char *friendly_name, char *configuration_file);
+mono_domain_create_appdomain (const char *friendly_name, char *configuration_file);
```
Comment (vargaz) [inline mono/metadata/appdomain.h]: Doesn't this change the api ?

Comment (lewurm) [inline mono/metadata/appdomain.h]: yes

**Comment by vargaz** [comment]
I don't think we should change the api, it _might_ break things.

**Comment by kumpera** [comment]
We can take the internal change, but not the API change, as it's a breaking change.

**Comment by lewurm** [comment]
An embedder was complaining about it because it makes their code ugly (that is, if you have strict coding style rules), but I understand the hesitation regarding API changes.

**Comment by vargaz** [comment]
If we have a chance to redo it we would make it const char*, but its too late for that.

**Pull request closed without merging**